packages/grafana-data/src/utils/extensions.ts

```diff
@@ -0,0 +1,134 @@
+import { SelectOptionItem } from '../types/select';
+
+export interface Extension {
+  id: string; // Unique Key -- saved in configs
+  name: string; // Display Name, can change without breaking configs
+  description: string;
+  aliasIds?: string[]; // when the ID changes, we may want backwards compatibility ('current' => 'last')
+
+  /**
+   * Some extensions should not be user selectable
+   *  like: 'all' and 'any' matchers;
+   */
+  excludeFromPicker?: boolean;
+}
+
+interface ExtensionSelectInfo {
+  options: Array<SelectOptionItem<string>>;
```

Comment:
Is it really needed to use the SelectOptionItem interface here? can't we define an ExtensionItem interface with things we need / use here? and then translate that to SelectOptionItem when we present the UI

Reply:
maybe... but we will want all the same fields, and the processing loop already happens here:
https://github.com/grafana/grafana/pull/17047/files#diff-e43a9d0466c5c7b68f77aefc4dab0762R81

but we could do it again.... or duck typing as you suggest